src/223/main/kotlin/org/rust/ide/hints/codeVision/RsImplementationsCodeVisionProvider.kt

```diff
@@ -0,0 +1,120 @@
+        }
+    }
+
+    override fun getHint(element: PsiElement, file: PsiFile): String? {
+        val rsElement = element as? RsElement ?: return null
+        if (rsElement !is RsTraitItem && rsElement !is RsAbstractable) return null
+
+        val implementationCount = element.implementationCount
+        if (implementationCount == 0) return null
+
+        return when (element) {
+            is RsTraitItem -> {
+                return RsBundle.message("rust.code.vision.implementation.hint", implementationCount)
+            }
+
+            is RsAbstractable -> {
+                return if (element.isAbstract) {
+                    RsBundle.message("rust.code.vision.implementation.hint", implementationCount)
+                } else {
+                    RsBundle.message("rust.code.vision.overrides.hint", implementationCount)
+                }
+            }
+
+            else -> null
+        }
+    }
+
+    override fun handleClick(editor: Editor, element: PsiElement, event: MouseEvent?) {
+        // copied from ImplsNavigationGutterIconRenderer::navigateToItems
+        if (event == null) return
+
+        val namedElement = element as? RsNamedElement ?: return
+        val elementName = namedElement.name ?: return
+
+        val targets = getImplementations(element).filterIsInstance<NavigatablePsiElement>().toTypedArray()
+        if (targets.isEmpty()) return
+
+        val escapedName = StringUtil.escapeXmlEntities(elementName)
+
+        PsiElementListNavigator.openTargets(
+            event,
+            targets,
+            CodeInsightBundle.message("goto.implementation.chooserTitle", escapedName, targets.size, ""),
+            CodeInsightBundle.message("goto.implementation.findUsages.title", escapedName, targets.size),
+            DefaultPsiElementCellRenderer()
+        )
+    }
+
+    override val relativeOrderings: List<CodeVisionRelativeOrdering>
+        get() = listOf(CodeVisionRelativeOrdering.CodeVisionRelativeOrderingAfter(RsReferenceCodeVisionProvider.ID))
+
+    override val id: String = ID
+
+    companion object {
+        const val ID = "rust.inheritors"
+    }
+}
+
+private fun getImplementationsQuery(element: PsiElement): Query<out PsiElement> {
+    return when (element) {
+        is RsTraitItem -> element.searchForImplementations()
+        is RsAbstractable -> element.searchForImplementations()
+        else -> EmptyQuery()
+    }
+}
+
+private fun countImplementations(element: PsiElement): Int = getImplementations(element).size
+
+private fun getImplementations(element: PsiElement): List<PsiElement> =
+    getImplementationsQuery(element).findAll().toList()
+
+private val IMPL_CACHE_KEY: Key<CachedValue<Int>> = Key.create("IMPL_CACHE_KEY")
+
+private val RsElement.implementationCount: Int
+    get() = CachedValuesManager.getCachedValue(this, IMPL_CACHE_KEY) {
+        val usages = countImplementations(this)
+        CachedValueProvider.Result.create(usages, PsiModificationTracker.MODIFICATION_COUNT)
```

Comment:
It doesn't seem it helps because cache with `PsiModificationTracker.MODIFICATION_COUNT` will drop after any code change

Reply:
Oh, this should have been the structure modification tracker, my bad. Thanks!